src/layoutBuilder.js

```diff
@@ -700,6 +760,66 @@ LayoutBuilder.prototype.buildNextLine = function (textNode) {
 		isForceContinue = inline.noNewLine && !isHardWrap;
 	}
 
+	if (textNode.rtl) {
```

Comment:
Let's tidy this logic up and update the comments (I won't remember what this does in 5 month's time).